gpu/preconditioner/block_jacobi_kernels.cu

```diff
@@ -166,6 +166,143 @@ GKO_ENABLE_IMPLEMENTATION_SELECTION(select_apply, apply);
 using compiled_kernels = syn::compile_int_list<1, 13, 16, 32>;
 
 
+template <typename IndexType>
+__global__ void compare_adjacent_rows(size_type num_rows, int32 max_block_size,
+                                      const IndexType *__restrict__ row_ptrs,
+                                      const IndexType *__restrict__ col_idx,
+                                      bool *matching_prev_row)
+{
+    const auto global_tid = blockDim.x * blockIdx.x + threadIdx.x;
+    const auto local_tid = threadIdx.x % cuda_config::warp_size;
+    const auto warp_id = global_tid / cuda_config::warp_size;
+
+    if (warp_id >= num_rows - 1) {
+        return;
+    }
+
+    const auto prev_row_start = row_ptrs[warp_id];
+    const auto curr_row_start = row_ptrs[warp_id + 1];
+    const auto next_row_start = row_ptrs[warp_id + 2];
+
+    const auto nz_this_row = next_row_start - curr_row_start;
+    const auto nz_prev_row = curr_row_start - prev_row_start;
+
+    if (nz_this_row != nz_prev_row) {
+        matching_prev_row[warp_id] = false;
```

Comment:
nit: `s/matching_prev_row/matching_next_row`?

Do I understand this correctly, `matching_prev_row[i]` is `true` if the `i`-th row and `i+1`-th row have the same pattern, or am I getting it wrong? Because if I'm not, wouldn't `matching_next_row` be a better name? As then we have `matching_next_row[i] == true` if `i`-th row matches the next (`i+1`-th) row.

Reply:
Agree, makes more sense. I just wanted to avoid "second_next_row_start" But I found a way...